### DISCUSSION TITLE
feat: Capacitor plugin wrapper around Flutter plugin

### DISCUSSION THREAD

**Discussion by timmyrosen** [ionic-team/capacitor#5313] [General]
### Description
Is it possible to create a Capacitor plugin wrapper around a Flutter plugin?

I have recently started to play around with Flutter and i'm a long time Ionic & Capacitor developer with a lot of existing apps build with these tools, and it would be amazing if I could create my own Capacitor plugins built with Flutter.

Maybe this isn't a feature request for Capacitor it self, but maybe someone here knows something about the topic.

### Platform(s)
iOS & Android

### Preferred Solution
To be able to call my plugin in the same way as existing Capacitor plugins, ie: MyFlutterPlugin.call() and await responses.

**Comment by mlynch** [comment]
Moving to discussions.  I should think this would be possible. We've had some success running Capacitor plugins in Cordova (going the opposite direction from Cordova -> Capacitor). At the end of the day they are just native classes with some methods. The issue is really in how much work it would be to wrap it for Capacitor.

This is near and dear to me right now because I'm exploring some ideas for bringing our Capacitor plugins to other ecosystems so I'll track the discussion here and follow up if we make any progress on this problem!

**Reply by timmyrosen** [reply]
Thanks for quick reply Max!

That sounds great. I think this would be an amazing feature, since there are so many great Flutter (and RN) plugins out there, and to be able to just fork them, maybe make a few adjustments, and create a Capacitor wrapper for it would be really powerful.

**Reply by codeisand** [reply]
please try this...because I cannot find any good wallpaper manager (Cordova/Capacitor) plugin for ionic.

**Comment by ghenry22** [comment]
@mlynch did this ever make any progress?  It would still be a great idea from a functionality perspective.

If it's not something that can be easily automated even a blog post covering the conventions and some examples would be cool as it would give people a starting point to start porting plugins when there is a gap in the capacity eco-system.

**Comment by lincolnthree** [comment]
Would also really love to see this so we can access some of that ecosystems functionality

**Reply by robingenz** [reply]
Which functionality are you missing? 😉
